src/clients/SpokePoolClient.ts

```diff
@@ -816,7 +827,7 @@ export class SpokePoolClient extends BaseAbstractClient {
 
       for (const event of fillEvents) {
         const fill = this.isV3FillEvent(event)
-          ? { ...(spreadEventWithBlockNumber(event) as V3FillWithBlock) }
+          ? { ...(spreadEventWithBlockNumber(event) as V3FillWithBlock), destinationChainId: this.chainId }
```

Comment:
@pxrl i think this was a bug before that was covered up by the `as` cast. Otherwise the new V3Fill doesn't get assigned a `destinationChain`

Reply:
Yeah this is a major pitfall of these type assertions :( Good catch - I will fix this in my PR.

Reply:
https://github.com/across-protocol/sdk-v2/pull/501/commits/960e6f01fc3aaaad6b8b9ff8aa76873845b24b14